tests/e2e.spec.ts

```diff
@@ -0,0 +1,149 @@
+  test.beforeEach(async ({ page }) => {
+    await loginAsGuest(page);
+    await page.goto('/create-quiz');
+  });
+
+  test('should render the create quiz page', async ({ page }) => {
+    expect(await page.title()).toMatch(/stwórz quiz/i);
+    expect(await page.locator('h1').textContent()).toMatch(/stwórz .* quiz/i);
+    expect(await page.getByPlaceholder(/tytuł quizu/i)).toBeVisible();
+    expect(await page.getByPlaceholder(/opis quizu/i)).toBeVisible();
+    expect(await page.locator('button', { hasText: /dodaj odpowiedź/i })).toBeVisible();
+    expect(await page.locator('button', { hasText: /dodaj pytanie/i })).toBeVisible();
+    expect(await page.locator('button', { hasText: /stwórz quiz/i })).toBeVisible();
+  });
+
+  test('should check if quiz title is not empty', async ({ page }) => {
+    await page.locator('button', { hasText: /stwórz quiz/i }).click();
+    expect(await page.locator('#container').getByText(/tytuł quizu/i)).toBeVisible();
+  });
+
+  test('should check if question content is not empty', async ({ page }) => {
+    await page.getByPlaceholder(/tytuł quizu/i).fill('Solvro Rulez');
+    await page.locator('button', { hasText: /stwórz quiz/i }).click();
+    expect(await page.locator('#container').getByText(/pytanie .* musi mieć treść/i)).toBeVisible();
+  });
+
+  test('should check if answer content is not empty', async ({ page }) => {
+    await page.getByPlaceholder(/tytuł quizu/i).fill('Solvro Rulez');
+    await page.getByRole('textbox', { name: 'Podaj treść pytania' }).fill('Czy lubisz Solvro?');
+    await page.locator('button', { hasText: /stwórz quiz/i }).click();
+    expect(await page.locator('#container').getByText(/odpowiedź .* musi mieć treść/i)).toBeVisible();
+  });
+
+  test('should create a quiz with valid data', async ({ page }) => {
+    await createProperQuiz(page);
+  });
+});
+
+test.describe('Quiz list', () => {
+  test.beforeEach(async ({ page }) => {
+    await loginAsGuest(page);
+    await page.goto('/quizzes');
+  });
+
+  test('should render the quiz list page', async ({ page }) => {
+    expect(await page.title()).toMatch(/twoje quizy/i);
+    expect(await page.locator('h1').textContent()).toMatch(/twoje quizy/i);
+    expect(await page.locator('button', { hasText: /stwórz quiz/i })).toBeVisible();
+  });
+
+  test('should show newly created quiz in the list', async ({ page }) => {
+    await createProperQuiz(page);
+    await page.goto('/quizzes');
+    await expect(page.locator(`text=${QUIZ_TITLE}`)).toBeVisible();
+  });
+
+  test('should show quiz details when clicking on a quiz', async ({ page }) => {
+    await createProperQuiz(page);
+    await page.goto('/quizzes');
+
+    const quiz = page.locator('div.card').first()
+    await expect(quiz).toBeVisible();
+    await expect(quiz).toContainText(QUIZ_TITLE);
+
+    const openQuizButton = quiz.getByRole('button', { name: /otwórz/i });
+    await openQuizButton.click();
+
+    await expect(page).toHaveURL(/\/quiz\/[a-z0-9-]+/);
```

Comment:
ja nie lubię za bardzo testowania urli, w sensie użytkownika nie bardzo to obchodzi na jakim url'u się znalazł tylko to czy widzi tytuł quizu